source/Grid/Grid.js

```diff
@@ -745,12 +752,14 @@ export default class Grid extends Component {
     // Gradually converging on a scrollTop that is within the bounds of the new, smaller height.
     // This causes a series of rapid renders that is slow for long lists.
     // We can avoid that by doing some simple bounds checking to ensure that scrollTop never exceeds the total height.
-    const { height, width } = this.props
+    const { autoHeight, height, width } = this.props
     const scrollbarSize = this._scrollbarSize
     const totalRowsHeight = this._rowSizeAndPositionManager.getTotalSize()
     const totalColumnsWidth = this._columnSizeAndPositionManager.getTotalSize()
     const scrollLeft = Math.min(Math.max(0, totalColumnsWidth - width + scrollbarSize), event.target.scrollLeft)
-    const scrollTop = Math.min(Math.max(0, totalRowsHeight - height + scrollbarSize), event.target.scrollTop)
+    const scrollTop = this.props.scrollTop && autoHeight
+      ? this.props.scrollTop
+      : Math.min(Math.max(0, totalRowsHeight - height + scrollbarSize), event.target.scrollTop)
```

Comment:
Gotcha. I think I'd probably rather document the limitation (for now) than do this. Thanks for proposing the option though. :)